onnxruntime/core/providers/js/operators/conv_transpose.h

```diff
@@ -0,0 +1,117 @@
+                                 static_cast<int32_t>(conv_transpose_attrs_.pads.size()),
+                                 static_cast<int32_t>(conv_transpose_attrs_.pads.size() > 1) ? conv_transpose_attrs_.pads[1] : 0,
+                                 static_cast<int32_t>(conv_transpose_attrs_.strides.size() > 0) ? conv_transpose_attrs_.strides[0] : 0,
+                                 static_cast<int32_t>(channels_last),
+                                 reinterpret_cast<int32_t>(&w_is_const_),
+                                 gsl::narrow_cast<int32_t>(conv_transpose_attrs_.output_shape.size()),
+                                 reinterpret_cast<int32_t>(conv_transpose_attrs_.output_padding.size() > 0 ? conv_transpose_attrs_.output_padding.data() : nullptr) >> 2,
+                                 gsl::narrow_cast<int32_t>(conv_transpose_attrs_.output_shape.size()),
+                                 reinterpret_cast<int32_t>(conv_transpose_attrs_.output_shape.size() > 0 ? conv_transpose_attrs_.output_shape.data() : nullptr) >> 2);
+    } else {
+      constexpr size_t pads_vec_size = 4;
+      constexpr size_t strides_vec_size = 2;
+      constexpr size_t dialations_vec_size = 2;
+      constexpr size_t kernel_shape_vec_size = 2;
+      // First set default values for pads, strides and dialations
+      std::vector<int32_t> local_pads(pads_vec_size, 0);
+      std::vector<int32_t> local_strides(strides_vec_size, 0);
+      std::vector<int32_t> local_dilations(dialations_vec_size, 0);
+      std::vector<int32_t> local_kernel_shape;
+      std::vector<int32_t> local_output_shape(conv_transpose_attrs_.output_shape.begin(), conv_transpose_attrs_.output_shape.end());
+      std::vector<int32_t> local_output_padding(conv_transpose_attrs_.output_padding.begin(), conv_transpose_attrs_.output_padding.end());
+      if (conv_transpose_attrs_.kernel_shape_specified) {
+        for (size_t i = 0; i < kernel_shape.size() && i < kernel_shape_vec_size; ++i) {
+          local_kernel_shape.push_back(kernel_shape[i]);
+        }
+      } else {
+        for (size_t i = 0; i < kernel_shape_vec_size; ++i) {
+          local_kernel_shape.push_back(0);
+        }
+      }
+      for (size_t i = 0; i < conv_transpose_attrs_.pads.size() && i < pads_vec_size; ++i) {
+        local_pads[i] = conv_transpose_attrs_.pads[i];
+      }
+      for (size_t i = 0; i < conv_transpose_attrs_.dilations.size() && i < dialations_vec_size; ++i) {
+        local_dilations[i] = conv_transpose_attrs_.dilations[i];
+      }
+      for (size_t i = 0; i < conv_transpose_attrs_.strides.size() && i < strides_vec_size; ++i) {
+        local_strides[i] = conv_transpose_attrs_.strides[i];
+      }
+      LOGS_DEFAULT(VERBOSE) << "output_shape = " << conv_transpose_attrs_.output_shape << std::endl;
+      LOGS_DEFAULT(VERBOSE) << "output_padding = " << conv_transpose_attrs_.output_padding << std::endl;
+      JSEP_INIT_KERNEL_ATTRIBUTE(ConvTranspose, ({
```

Comment:
I feel sorry to make you have to write those. I am thinking about how to do this more simple